web/src/app/services/data-loader.service.ts

```diff
@@ -393,37 +393,40 @@ export class InspectionDataLoaderService {
     return result;
   }
 
-  public async loadInspectionDataFromBackend(taskId: string) {
+  public async loadInspectionDataFromBackend(inspectionID: string) {
     this.progress.show();
     this.progress.updateProgress({
       message: 'Downloading inspection data...',
       percent: 0,
       mode: 'determinate',
     });
-    const metadata = await lastValueFrom(
-      this.backendService.getInspectionMetadata(taskId),
-    );
-    const allSize = metadata.header.fileSize ?? 0;
-    const data = await lastValueFrom(
-      this.backendService.getInspectionData(taskId, (done) => {
-        this.progress.updateProgress({
-          message: `Downloading inspection data...(${ProgressUtil.formatPogressMessageByBytes(
-            done,
-            allSize,
-          )})`,
-          percent: (done / allSize) * 100,
-          mode: 'determinate',
-        });
-      }),
-    );
-    this.progress.dismiss();
-    if (data === null) {
+    try {
+      const data = await lastValueFrom(
+        this.backendService.getInspectionData(
+          inspectionID,
+          ((acc) => (allSize, done) => {
+            acc += done;
+            this.progress.updateProgress({
+              message: `Downloading inspection data...(${ProgressUtil.formatPogressMessageByBytes(
+                acc,
+                allSize,
+              )})`,
+              percent: (acc / allSize) * 100,
+              mode: 'determinate',
+            });
+          })(0),
```

Comment:
Accumulating done bytes in ProgressReporter would be better to have done in `getInspectionData`?
I think the `DownloadProgressReporter` should naturally be called with the done bytes of entire file instead of a chunk sizes used in the func because the parallel behavior is hidden in the func.